### PR TITLE
Move connect:state and connect:info to co blocks

### PR DESCRIPTION
**Pull request by gulopine**

### CODE REVIEW

File: lib/commands/connect/info.js [hunk 1]
```diff
@@ -23,18 +14,22 @@ module.exports = {
...
-          } else {
-            connections_info(connections);
-          }
+    run: cli.command(co.wrap(function* (context, heroku) {
+      let connections = yield api.withUserConnections(context.auth.password, context.app, context.flags, true, heroku);
+
+      if (connections.length === 0) {
+        console.log("No connection found, requesting auth...");
+        yield api.requestAppAccess(context.auth.password, context.app);
+        let connections = yield api.withUserConnections(context.auth.password, context.app, context.flags, false, heroku);
```
Comment (lambacck) [inline lib/commands/connect/info.js]: Let here is not what you want. It won't escape the if condition.

Comment (gulopine) [inline lib/commands/connect/info.js]: Ah, yes. Previously, I had kept the dual connections_info calls, so it didn't need to escape it. Didn't catch this when I tidied it up.

File: lib/commands/connect/info.js [hunk 2]
```diff
@@ -23,18 +14,22 @@ module.exports = {
...
-            api.requestAppAccess(context.auth.password, context.app).then(function() {
-              api.withUserConnections(context.auth.password, context.app, context.flags, false, heroku).then(function(connections){
-                connections_info(connections);
-              });          
-            });
-          } else {
-            connections_info(connections);
-          }
+    run: cli.command(co.wrap(function* (context, heroku) {
+      let connections = yield api.withUserConnections(context.auth.password, context.app, context.flags, true, heroku);
```
Comment (lambacck) [inline lib/commands/connect/info.js]: I'd probably have this be a var, since the scope is the function and then it's more clear what is happening.

**Comment by lambacck** [comment]
Other than var vs let which is a style nit :+1:

File: lib/commands/connect/info.js (hunk 2 above)
Comment (gulopine) [inline lib/commands/connect/info.js]: Yeah, I'm not quite sure what current JS style would indicate. The Heroku toolbelt plugin docs use `let` even for top-level requires, as you see at the top of this file as well, so I've been following suit. I don't have any opinion myself, though.

**Pull request merged**